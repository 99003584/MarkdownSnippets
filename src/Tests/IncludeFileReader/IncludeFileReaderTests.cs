using System;
using System.Collections.Generic;
using Xunit;
using Xunit.Abstractions;

public class IncludeFileReaderTests :
    XunitApprovalBase
{
    [Fact]
    public void Simple()
    {
        var files = IncludeFileReader.ReadIncludes(
            new List<string>
            {
                @"IncludeFileReader\file.include.md"
            });
        ObjectApprover.Verify(files("file"));
    }

    [Fact]
    public void Missing()
    {
        var files = IncludeFileReader.ReadIncludes(new List<string>());
        var exception = Assert.Throws<Exception>(() => files("missing"));
        ObjectApprover.Verify(exception.Message);
    }

    public IncludeFileReaderTests(ITestOutputHelper output) :
        base(output)
    {
    }
}