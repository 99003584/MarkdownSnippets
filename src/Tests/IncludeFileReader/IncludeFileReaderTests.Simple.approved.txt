{
  Error: '',
  Key: 'file',
  Path: 'IncludeFileReader\file.include.md',
  Lines: [
    'Simple1'
  ]
}